### PR TITLE
tracing: xray annotation documentation and CODEOWNER

### PR DESCRIPTION
**Pull request by nbaws**
Commit Message: xray annotation documentation and CODEOWNER
Additional Description: minor amendment to xray documentation and include @nbaws as xray CODEOWNER
Risk Level: N/A
Testing: N/A
Docs Changes: Yes
Release Notes:
Platform Specific Features:
[Optional Runtime guard:]
[Optional Fixes #Issue] https://github.com/envoyproxy/envoy/issues/35147
[Optional Fixes commit #PR or SHA]
[Optional Deprecated:]
[Optional [API Considerations](https://github.com/envoyproxy/envoy/blob/main/api/review_checklist.md):]

### CODE REVIEW

File: CODEOWNERS [hunk 1]
```diff
@@ -62,7 +62,7 @@ extensions/filters/common/original_src @klarose @mattklein123
 # tracers.datadog extension
 /*/extensions/tracers/datadog @dmehala @mattklein123
 # tracers.xray extension
-/*/extensions/tracers/xray @suniltheta @mattklein123
+/*/extensions/tracers/xray @suniltheta @mattklein123 @nbaws
```
Comment (adisuissa) [inline CODEOWNERS:65]: cc @suniltheta to approve

File: api/envoy/config/trace/v3/xray.proto [hunk 2]
```diff
@@ -21,7 +21,63 @@ option (udpa.annotations.file_status).package_version_status = ACTIVE;
 
 // [#protodoc-title: AWS X-Ray Tracer Configuration]
 // Configuration for AWS X-Ray tracer
-
+// The X-Ray tracer will automatically attach :ref:`custom_tags <envoy_v3_api_field_extensions.filters.network.http_connection_manager.v3.HttpConnectionManager.Tracing.custom_tags>` as annotations to the span.
+// See: `Annotations https://docs.aws.amazon.com/xray/latest/devguide/xray-concepts.html#xray-concepts-annotations`
```
Comment (adisuissa) [inline api/envoy/config/trace/v3/xray.proto:25]:
```suggestion
// See: `Annotations <https://docs.aws.amazon.com/xray/latest/devguide/xray-concepts.html#xray-concepts-annotations>`_.
```

**Review by adisuissa — COMMENTED** [review]
Overall LGTM, modulo the nit.
@suniltheta to approve code-ownership of the xray extension.

**Review by suniltheta — APPROVED** [review]

**Review by adisuissa — APPROVED** [review]
LGTM, thanks!
/lgtm api

**Comment by yanavlasov** [comment]
@nbaws please merge main to get CI unstuck.

**Review by adisuissa — APPROVED** [review]
/lgtm api
LGTM, thanks!

**Pull request merged**